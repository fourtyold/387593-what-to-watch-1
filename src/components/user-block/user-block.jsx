import React from "react";
import PropTypes from "prop-types";
import {Link} from "react-router-dom";
import {connect} from "react-redux";

import {getAvatarUrl} from "../../reducer/user/selectors.js";
import {Operation} from "../../reducer/films/films.js";

const UserBlock = (props) => {
  const {avatarUrl, getFavoritesList} = props;
  if (avatarUrl) {
    return <div className="user-block__avatar" onClick={getFavoritesList}>
      <Link to="/mylist">
        <img src={`https://es31-server.appspot.com${avatarUrl}`} alt="User avatar" width="63" height="63" />
      </Link>
    </div>;
  }
  return <Link to="/login" className="user-block__link">Sign in</Link>;
};

UserBlock.propTypes = {
  avatarUrl: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]),
  getFavoritesList: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  getFavoritesList: () => {
    dispatch(Operation.getFavoritesList());
  }
});

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  avatarUrl: getAvatarUrl(state)
});

export {UserBlock};

export default connect(mapStateToProps, mapDispatchToProps)(UserBlock);
