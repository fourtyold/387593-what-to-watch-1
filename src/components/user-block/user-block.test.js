import React from "react";
import renderer from "react-test-renderer";
import {HashRouter} from "react-router-dom";

import {UserBlock} from "./user-block.jsx";

const options = {
  avatarUrl: `avatar`,
  getFavoritesList: () => {}
};

it(`User block renders correctly`, () => {
  const tree = renderer.create(
      <HashRouter>
        <UserBlock
          avatarUrl={options.avatarUrl}
          getFavoritesList={options.getFavoritesList}
        />
      </HashRouter>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
