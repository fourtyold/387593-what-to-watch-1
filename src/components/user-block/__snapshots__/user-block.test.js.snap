// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User block renders correctly 1`] = `
<div
  className="user-block__avatar"
  onClick={[Function]}
>
  <a
    href="#/mylist"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="https://es31-server.appspot.comavatar"
      width="63"
    />
  </a>
</div>
`;
